### ISSUE TITLE
Spotify client main section ignoring colorscheme

### ISSUE THREAD

**Issue by GideonWolfe** [themix-project/oomoxify#5]
So I am writing a script to theme my system based on a wallpaper, and part of that script is to call `oomoxify-cli` to theme spotify.

The line in my script is as follows:

`oomoxify-cli  $HOME/.cache/wal/colors-oomox -s /opt/spotify/Apps/`

On certain pictures, all looks as expected with the spotify client adopting mostly the same color. However on most images, the middle part of the spotify client is mostly left alone.
![DeepinScreenshot_select-area_20190726183106](https://user-images.githubusercontent.com/32942052/61988482-bf5e4800-afd6-11e9-94af-71f061a35f13.png)

This looks a bit strange.

**Comment by actionless** [comment]
please use oomox-gui instead, that could be bug in wal

**Comment by GideonWolfe** [comment]
@actionless

The whole point of my script is to *not* use the GUI.

Isn't running `oomoxify-cli` the exact same thing as clicking `Apply Spotify Theme` in the GUI? If not, why is that so?

I don't think this is a bug with Wal. Wal chooses the colors fine, I'm simply asking that oomox chooses the *same* color for the whole interface.

**Comment by actionless** [comment]
so at least as debugging points open ui, choose spotify colors in the bottom of middle pane, and export and compare the result

also you can use theme created from GUI in the cli, it's saved in ~/.config/oomox/colors

**Comment by GideonWolfe** [comment]
@actionless So I ran a `diff` command on both the themes created by the GUI and from the CLI, and there is no differences between the files.

And I am not seeing the xresources theme inside `~/.config/oomox/colors`

**Comment by actionless** [comment]
have you done this step `choose spotify colors in the bottom of middle pane` ?

**Comment by GideonWolfe** [comment]
The `xresources2` theme says it "Can't be edited in the GUI". When I try to change the settings in the colors-oomox theme to make the spotify foreground the same as the background, the entire app becomes unreadable and black.

There are only three colors, `foreground`, `background` and `accent`. They should all be different so I can see things.

**Comment by actionless** [comment]
> The xresources2 theme says it "Can't be edited in the GUI".

so try the one which not saying that

**Comment by GideonWolfe** [comment]
> When I try to change the settings in the colors-oomox theme

As I said in my last comment, I did try that, with the `colors-oomox` theme. This theme is editable in the GUI, and manually changing the spotify settings in the middle window does not improve the issue.

**Comment by actionless** [comment]
i even double-checked what english is your native language, but i still can't get from your description what's not working for you, idk mb try using 1-2-3) points and more screenshots of how you do steps in gui app and they leading you to some result

**Comment by GideonWolfe** [comment]
I thought I described it pretty clearly, but ok. I am writing a [theming tool](https://github.com/GideonWolfe/autoTheme/) which automatically applies a system wide theme using some of the oomox command line options, making it so you don't need any GUI to change themes.

Here is how the issue happens.

First, i run `wal -i path/to/image` to set my terminal colorscheme. This outputs the colorscheme to my `.Xresources` file.

When I open `oomox-gui`, I see a drop down theme section named `Xresources:xresources` that contain two variants of the theme generated by `wal`.

![DeepinScreenshot_select-area_20190728132622](https://user-images.githubusercontent.com/32942052/62012456-547a5180-b13b-11e9-9ecd-95c05b7acc65.png)
Note that I am unable to manually edit this color scheme in the GUI, which is not an issue because I want to stay away from the GUI.

I want to use one of these themes, as they will match with my terminal and any other program using `.Xresources` as color palette.

I can apply the GTK theme successfully with the following command:

`oomox-cli /opt/oomox/scripted_colors/xresources/xresources2`

And this looks great.

I can update my icon theme with the following command

`oomox-archdroid-icons-cli /opt/oomox/scripted_colors/xresources/xresources2`

And they look fine.

However, when I try to run the following command

`oomoxify-cli  $HOME/.cache/wal/colors-oomox -s /opt/spotify/Apps/`

to try and apply the wal theme to spotify, I expect an output like 

![image](https://user-images.githubusercontent.com/32942052/62012475-a3c08200-b13b-11e9-92a3-d3908484dcb3.png)

(this was made with spicetify because I couldnt find a screenshot of the oomoxify as it should look.)

Note how the app is one solid color. Instead, I get a theme that looks like this

![DeepinScreenshot_select-area_20190726183106](https://user-images.githubusercontent.com/32942052/62012480-bdfa6000-b13b-11e9-92f0-b34f4da5425b.png)

where the side menu bar, play controls and header look fine, but the entire center of the app is black and unthemed.

Like you said, I tried to edit the `colors-oomox` theme, even though this isnt the one I want to use (it is pretty close to the .Xresources theme though). The spotify color options in the bottom right of the middle pane look like how you would expect them to look from the wal colorshceme generated. 

![DeepinScreenshot_select-area_20190728133106](https://user-images.githubusercontent.com/32942052/62012502-fdc14780-b13b-11e9-9a2a-ca7b797312b2.png)

I use super dark colorschemes usually, so I expect the colors to be dark. One reason why the middle area looks unthemed is because it's just way darker than the rest for some reason. When I tried it with a moderately light theme, I still get strange artifacts even though it is taking some colors from my theme.

![DeepinScreenshot_select-area_20190728133900](https://user-images.githubusercontent.com/32942052/62012589-14b46980-b13d-11e9-8738-b419f8d4a3e6.png)

**Comment by actionless** [comment]
but it's not unthemed/black, it's shaded background color, because in original spotify those areas also not in the same color

you could fine-tune every block by setting additional `SPOTIFY_*` keys in the theme file, see the key names here:
https://github.com/themix-project/oomoxify/blob/master/oomoxify.sh#L111-L140


btw how do you think, will it be useful making a cli entrypoint for oomox own's color import from images? (like on this pic https://raw.githubusercontent.com/themix-project/oomox/master/screenshots/pokedex_dawn.png) 

i did created a ticket for discussion on that: https://github.com/themix-project/oomox/issues/202 but so far it have no interest

**Comment by GideonWolfe** [comment]
I personally think that would be a great feature! It's always good to have options, especially if you want to automate the theming in a script like me

**Comment by actionless** [comment]
the thing is what after playing for few years with automating it with scripts i've realized what there is no one preset which will fit all possible source pictures/color palettes, so it's always better to visually choose from couple of possible options

so for me hard to imagine what would be the most demanded from users in such cli, so please don't hesitate to leave some suggestions in that ticket (https://github.com/themix-project/oomox/issues/202) or tell there for which GUI-only thing you wanna see the CLI first

**Comment by GideonWolfe** [comment]
> i've realized what there is no one preset which will fit all possible source pictures/color palettes

I totally agree with that, which is why tools like oomox and wpgtk are great, they allow you to fine tune things if necessary.

The main point of my script is to hook into external programs (such as a plugin for wal that themes steam, or vscode) that would otherwise need to be manually run every time a new theme is applied. 

One thing I would like to see in the CLI is a choice between light and dark themes generated by .Xresources (although I don't see this option in the GUI either). This screenshot can explain what I mean:
![DeepinScreenshot_select-area_20190726183106](https://user-images.githubusercontent.com/32942052/62020804-464b2600-b179-11e9-9de2-60d6d94cd9aa.png)

My terminal is a mainly dark theme, with light accents. The GTK generated by oomox definitely uses colors from the same palette, but is almost inverse (dark icons and text on a light background). It would be nice to see a light and dark option :)

If someone like myself knew that they always wanted to use a dark theme, they could change their script to say `oomox-cli /opt/oomox/scripted_colors/xresources/xresources2-dark`

**Comment by actionless** [comment]
yeah, feel free to contribute more xresources-based themes:

https://github.com/themix-project/oomox/tree/master/colors/Xresources
https://github.com/themix-project/oomox/tree/master/scripted_colors/xresources

**Comment by GideonWolfe** [comment]
Ok, I have a dark theme that seems to be working!

https://gist.github.com/GideonWolfe/60e57f9a1a2b2351a02e3290fda44be0

![DeepinScreenshot_select-area_20190726183106](https://user-images.githubusercontent.com/32942052/62022805-c32ecd80-b182-11e9-93b2-b7b7414b759b.png)

I can test it out to make sure there are no weird issues with other colors/gtk menus and make a PR

**Comment by actionless** [comment]
yup, thanks for an update!

**Comment by GideonWolfe** [comment]
So I ran into another issue while I was testing my dark theme. I can make a new ticket or we can continue here. I may be misunderstanding something, but here are the steps to reproduce for me:

1. Copy file `/opt/oomox/colors/Xresources/xresources3` to a new file `xresources-dark` in the same directory. Copy `opt/oomox/scripted_colors/xresources/xresources3` to `opt/oomox/scripted_colors/xresources/xresources-dark`

2. To test this script, I would edit the color values in the `xresources-dark` file, and open up the GUI to visually inspect the theme. To test on other programs, I ran "Export Theme" and "Export Icons" from the GUI. After a few tries, I had a theme I was ready to test on more pictures. (this is where I sent the last screenshot)

3. This is where I ran into my issue. I tried incorporating the dark theme into my script so I could see how it works with other wallpapers. The command I ran in my script:

`oomox-cli /opt/oomox/scripted_colors/xresources/xresources-dark` and 
`oomox-archdroid-icons-cli /opt/oomox/scripted_colors/xresources/xresources-dark`

When I call oomox from the command line like this, I get the *wrong* look. Compare the GTK theme exported from the command line with the version exported from the GUI using the *exact* same source file: After I ran the above commands, I opened oomox GUI

![DeepinScreenshot_select-area_20190729151846](https://user-images.githubusercontent.com/32942052/62086747-346a9100-b214-11e9-9e35-6ad3fbd04575.png)

I then close and reopen the GUI, and the preview on the right looks how I would expect (with a dark background)
![DeepinScreenshot_select-area_20190729152016](https://user-images.githubusercontent.com/32942052/62086806-6845b680-b214-11e9-85b5-4e674c38cb47.png)

Only after I click the "Export Theme" button does my theme apply sucsessfully.
![DeepinScreenshot_select-area_20190729152134](https://user-images.githubusercontent.com/32942052/62086861-95926480-b214-11e9-8c2f-39b4cb296972.png)

I suspect this might have something to do with the file at `opt/oomox/scripted_colors/xresources/xresources-dark`. I got my dark theme working without editing this file at all, only `/opt/oomox/colors/Xresources/xresources-dark`. However, running the command seems to ignore the dark theme, and instead generates what is basically `xresources3`. To test, I switched the FG and BG values in `/opt/oomox/scripted_colors/xresources-dark`, which should approximate the dark theme. This didn't work :(

**Comment by actionless** [comment]
but themes from `scripted_colors` are not showing up in UI

**Comment by GideonWolfe** [comment]
Interesting. During testing, I would edit the scripted_colors file and it would be reflected in the GUI theme preview. Only when I run `oomox-cli` does it not show up correctly

**Comment by GideonWolfe** [comment]
Any update on this? When I run `oomox-cli /opt/oomox/scripted_colors/xresources/xresources-dark` I do not successfully apply/generate a dark colorscheme. It stays light.

When I select `xresource-dark` from the GUI and select `Export Theme` it works fine though.

**Comment by actionless** [comment]
> Any update on this?

idk what kind of update do you expect since i've already mentioned what 

> themes from scripted_colors are not showing up in UI

so please double-check the steps you're doing

**Comment by actionless** [comment]
or, let me rephrase, themes in `scripted_colors` dir in totally different format than the "normal" themes in `colors` dir, ie the ones in scripted_colors are actually very tiny bash scripts

**Comment by GideonWolfe** [comment]
I understand that the scripted colors do not show up in the GUI. I can delete the `xresources-dark` file in `scripted_colors` entirely and the theme will still generate from .Xresources and show up in the GUI.

But if I delete this file, I have no file to run `oomox-cli` on. If I'm correct, you need to point `oomox-cli` and `oomox-archdroid-icons-cli` towards a file in `scripted_colors`. My theming script relies on using these commands, but no matter what values I try to change in the `scripted_colors` file, it does not output the same as in the GUI. is there a certain value I need to set to be able to change the color values in this file?

EDIT: looks like messing with a few more values got the results I was looking for. I will include both files in my PR

**Comment by actionless** [comment]
btw i think name `xresources-reverse` would make more sense than `dark`, since you can't know in advance is it gonna be dark or not
